src/abstractdatatable/abstractdatatable.jl

```diff
@@ -542,35 +535,13 @@ function Base.convert{T}(::Type{Matrix{T}}, dt::AbstractDataTable)
     res = Matrix{T}(n, p)
     idx = 1
     for (name, col) in zip(names(dt), columns(dt))
-        any(isnull, col) && error("cannot convert a DataTable containing null values to array (found for column $name)")
+        !(T >: Null) && any(isnull, col) && error("cannot convert a DataTable containing null values to array (found for column $name)")
```

Comment:
Ah, sorry, I thought this was just a fast path to avoid checking for nulls inside a non-nullable array.